event_test.go

```diff
@@ -83,7 +83,10 @@ func TestUnmarshalEvent(t *testing.T) {
 }
 
 func TestUnmarshalEvents(t *testing.T) {
-	origRegisteredEvents := registeredEvents
+	origRegisteredEvents := map[string]Event{}
```

Comment:
While this was certainly a bug (good catch!), I would solve it differently: In line 91-93 (original) the map gets cleared entirely. So we can just do:
```
origRegisteredEvents := registeredEvents
defer func() {
	registeredEvents = origRegisteredEvents
}()
registeredEvents = make(map[string]Event)
```